PhysicsTools/PatAlgos/python/slimming/puppiForMET_cff.py

```diff
@@ -21,52 +18,34 @@ def makePuppies( process ):
-    process.puppiForMET.useRefs          = False
-    #Line below points puppi MET to puppi no lepton which increases the response
-    process.puppiForMET.puppiCandName    = 'puppiMerged'
-
+    addToProcessAndTask('puppiNoLepPhoton', process.puppiNoLep.clone(), process, task)
+    process.puppiNoLepPhoton.PtMaxPhotons = cms.double(20.)
+    process.puppiForMET = cms.EDProducer("CandViewMerger",src = cms.VInputTag( 'puppiNoLepPhoton','pfLeptonsPUPPET'))
+    task.add(process.puppiForMET)
 
 def makePuppiesFromMiniAOD( process, createScheduledSequence=False ):
     task = getPatAlgosToolsTask(process)
     process.load('CommonTools.PileupAlgos.Puppi_cff')
     task.add(process.puppi)
     process.puppi.candName = cms.InputTag('packedPFCandidates')
-    process.puppi.clonePackedCands = cms.bool(True)
+    process.puppi.clonePackedCands = True
     process.puppi.vertexName = cms.InputTag('offlineSlimmedPrimaryVertices')
-    process.puppi.useExistingWeights = cms.bool(True)
+    process.puppi.useExistingWeights = True
     process.pfNoLepPUPPI = cms.EDFilter("CandPtrSelector", src = cms.InputTag("packedPFCandidates"), cut =  cms.string("abs(pdgId) != 13 && abs(pdgId) != 11 && abs(pdgId) != 15"))
     task.add(process.pfNoLepPUPPI)
     process.pfLeptonsPUPPET   = cms.EDFilter("CandPtrSelector", src = cms.InputTag("packedPFCandidates"), cut = cms.string("abs(pdgId) == 13 || abs(pdgId) == 11 || abs(pdgId) == 15"))
     task.add(process.pfLeptonsPUPPET)
     addToProcessAndTask('puppiNoLep', process.puppi.clone(), process, task)
-    process.puppiNoLep.candName = cms.InputTag('pfNoLepPUPPI') 
-    process.puppiNoLep.useWeightsNoLep = cms.bool(True)
-    process.puppiNoLep.useExistingWeights = cms.bool(True)
-    process.puppiMerged = cms.EDProducer("CandViewMerger",src = cms.VInputTag( 'puppiNoLep','pfLeptonsPUPPET'))
-    task.add(process.puppiMerged)
-    process.load('CommonTools.PileupAlgos.PhotonPuppi_cff')
-    task.add(process.puppiPhoton)
-    addToProcessAndTask('puppiForMET', process.puppiPhoton.clone(), process, task)
-    process.puppiForMET.candName = cms.InputTag('packedPFCandidates')
-    process.puppiForMET.photonName = cms.InputTag('slimmedPhotons')
-    process.puppiForMET.runOnMiniAOD = cms.bool(True)
-    setupPuppiPhotonMiniAOD(process)
-    task.add(process.egmPhotonIDTask)
-    #Line below replaces reference linking wiht delta R matching because the puppi references after merging are not consistent with those of the original packed candidate collection
-    process.puppiForMET.useRefs          = False
-    #Line below points puppi MET to puppi no lepton which increases the response
-    process.puppiForMET.puppiCandName    = 'puppiMerged'
+    process.puppiNoLep.candName = cms.InputTag('pfNoLepPUPPI')
+    process.puppiNoLep.useWeightsNoLep = True
+    addToProcessAndTask('puppiNoLepPhoton', process.puppiNoLep.clone(), process, task)
+    process.puppiNoLepPhoton.PtMaxPhotons = cms.double(20.)
```

Comment:
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.